### DISCUSSION TITLE
Changing Document Alias does not re-generate Generated Types

### DISCUSSION THREAD

**Discussion by Nisden** [JasperFx/marten#2567] [General] [answered]
When using `.OptimizeArtifactWorkflow()` with Visual Studio, and you modify the DocumentAlias of an document that have already had a Provider generated, the Provider is not updated to reflect the new function names.

This cause SQL exceptions where its complaining about missing the function `mt_upsert_(OLD MODEL NAME)`

**Comment by jeremydmiller** [comment, marked as answer]
Yeah, sorry man, there's no automatic detection of that. `dotnet run -- codegen delete` && `dotnet run -- codegen write` is your only alternative here. Or manually delete the files. It's a known limitation in both Marten and now Wolverine. 

I'm moving this to a discussion as a future direction kind of thing.

**Reply by Nisden** [reply]
For now I have just changed development to "Dynamic" and then use our CI/CD to create the codegen and switch to "Static".
The "Auto" mode just seems full of pitfalls.

Maybe this should be documented some where? Auto feel like it should just be auto, but its not.
